drivers/s3.go

```diff
@@ -372,6 +373,22 @@ func (os *s3Session) saveDataPut(ctx context.Context, name string, data io.Reade
 	return url, nil
 }
 
+func (os *s3Session) Delete(ctx context.Context, name string) error {
```

Comment:
I see this method is not in the main object store interface. I converted it manually on the hack script where I used it. We can either add it to the main interface and return "not implemented" for other drivers, or even make the "object store deleter" public but still optional and have clients convert stuff to check capabilities. wdyt?